### PR TITLE
dry-run: Add DryRunAll value

### PR DESCRIPTION
**Pull request by apelisse**
Re-open of #66139 that ended-up being broken ?

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by neolit123** [comment]
@kubernetes/sig-api-machinery-pr-reviews

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 1]
```diff
@@ -418,6 +418,11 @@ const (
 	DeletePropagationForeground DeletionPropagation = "Foreground"
 )
 
+const (
+	// DryRunTrue means that all dry run stages will be processed.
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: Which means? Everything but the last etcd storage call? Missing a definition what dry-run stages are.

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go [hunk 2]
```diff
@@ -84,6 +85,25 @@ func ValidateDeleteOptions(options *metav1.DeleteOptions) field.ErrorList {
 		*options.PropagationPolicy != metav1.DeletePropagationOrphan {
 		allErrs = append(allErrs, field.NotSupported(field.NewPath("propagationPolicy"), options.PropagationPolicy, []string{string(metav1.DeletePropagationForeground), string(metav1.DeletePropagationBackground), string(metav1.DeletePropagationOrphan), "nil"}))
 	}
+	allErrs = append(allErrs, validateDryRun(options.DryRun)...)
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go]: for consistency I would create a NewPath here and pass it as fldPath.

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go (hunk 1 above)
Comment (apelisse) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: To be honest, we don't know yet what the stages are, and don't really have a proposal for what they might be. I agree that this should be rephrased.

Comment (sttts) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: What about just writing: DryRunTrue means to execute the complete REST handler with all stages, but don't store the result to etcd.

Comment (apelisse) [reply staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: Sounds good to me!

**Comment by apelisse** [comment]
@liggitt @sttts PTAL

**Comment by sttts** [comment]
sgtm.

**Comment by apelisse** [comment]
@kubernetes/sig-api-machinery-api-reviews any objection to using True as the only possible value for dryRun (for now), as opposed to * or All?

**Comment by apelisse** [comment]
> @kubernetes/sig-api-machinery-api-reviews any objection to using True as the only possible value for dryRun (for now), as opposed to * or All?

@kubernetes/api-reviewers

**Comment by smarterclayton** [comment]
If there are ever going to be more than one enumeration value, `All` is more appropriate.  `True` is very much boolean, implying `False` and possibly `Unknown` (like condition).  The point of this flag is to potentially support subset of stages.  So `All` is far more appropriate than `True` in this context of an enum or a selection out of a set.

**Comment by liggitt** [comment]
> All is far more appropriate than True in this context of an enum or a selection out of a set.

works for me

**Pull request merged**